src/dash/utils/TimelineSegmentsGetter.js

```diff
@@ -48,7 +48,13 @@ function TimelineSegmentsGetter(config, isDynamic) {
         var list = base.SegmentURL_asArray;
         var isAvailableSegmentNumberCalculated = representation.availableSegmentsNumber > 0;
 
-        var maxSegmentsAhead = 10;
+        let maxSegmentsAhead;
+        if (availabilityUpperLimit) {
+            maxSegmentsAhead = availabilityUpperLimit
+        } else {
+            maxSegmentsAhead = (index > -1 || requestedTime !== null) ? 10 : Infinity;
```

Comment:
dash.js is full of magic numbers, probably we should start fixing this by declaring constants, ie extracting `10` to a constant in this case.